### ISSUE TITLE
Permission denied errors

### ISSUE THREAD

**Issue by ddriddle** [gruntwork-io/terragrunt#225]
> $ terragrunt --version
terragrunt version v0.12.16

>$ terraform --version
Terraform v0.9.5

> $ git --version
git version 2.1.4

> $cat /etc/debian_version
8.6

> $ umask
0022

I am receiving a file permission denied error when I run `terragrunt plan` after some state movement. The 'terragrunt state mv' commands are also receiving permission denied errors. The first time I ran plan it succeeded but not after the state changes. The error message that I am receiving is:
```
[terragrunt] 2017/05/25 08:40:44 open /tmp/terragrunt-download/nh-GXY9InsiG9x9QeDXnxdMwETM/YNBFcR4X61Vu2vYlFhZFR15I3qI/as-enterprise-vpc/.terraform/modules/7f243cf8f80c562d50d2599756bd5ad5/.git/objects/pack/pack-b9a3ac6087e03268b21dc694d20ccf89859a98d3.idx: permission denied
[terragrunt] 2017/05/25 08:40:44 Unable to determine underlying exit code, so Terragrunt will exit with error code 1
```
Here are the file perms for that file:
```
$ ls -l /tmp/terragrunt-download/nh-GXY9InsiG9x9QeDXnxdMwETM/YNBFcR4X61Vu2vYlFhZFR15I3qI/as-enterprise-vpc/.terraform/modules/7f243cf8f80c562d50d2599756bd5ad5/.git/objects/pack/pack-b9a3ac6087e03268b21dc694d20ccf89859a98d3.idx
-r--r--r-- 1 ddriddle ddriddle 1996 May 25 08:40 /tmp/terragrunt-download/nh-GXY9InsiG9x9QeDXnxdMwETM/YNBFcR4X61Vu2vYlFhZFR15I3qI/as-enterprise-vpc/.terraform/modules/7f243cf8f80c562d50d2599756bd5ad5/.git/objects/pack/pack-b9a3ac6087e03268b21dc694d20ccf89859a98d3.idx
```
Here is the first plan I ran which succeeded:
```
$ terragrunt plan
[terragrunt] [/home/ddriddle/src/as-aws/as-aws-test/us-east-2/vpc] 2017/05/25 08:40:03 Running command: terraform --version
[terragrunt] 2017/05/25 08:40:04 Reading Terragrunt config file at /home/ddriddle/src/as-aws/as-aws-test/us-east-2/vpc/terraform.tfvars
[terragrunt] 2017/05/25 08:40:04 Downloading Terraform configurations from git::ssh://git@github.com/cites-illinois/as-aws-modules.git?ref=v0.5 into /tmp/terragrunt-download/nh-GXY9InsiG9x9QeDXnxdMwETM/YNBFcR4X61Vu2vYlFhZFR15I3qI
[terragrunt] 2017/05/25 08:40:04 Running command: terraform init -backend=false -get=false git::ssh://git@github.com/cites-illinois/as-aws-modules.git?ref=v0.5 /tmp/terragrunt-download/nh-GXY9InsiG9x9QeDXnxdMwETM/YNBFcR4X61Vu2vYlFhZFR15I3qI
Initializing configuration from: "git::ssh://git@github.com/cites-illinois/as-aws-modules.git?ref=v0.5"...
Terraform initialized in an empty directory!

The directory has no Terraform configuration files. You may begin working
with Terraform immediately by creating Terraform configuration files.
[terragrunt] 2017/05/25 08:40:06 Copying files from /home/ddriddle/src/as-aws/as-aws-test/us-east-2/vpc into /tmp/terragrunt-download/nh-GXY9InsiG9x9QeDXnxdMwETM/YNBFcR4X61Vu2vYlFhZFR15I3qI/as-enterprise-vpc
[terragrunt] 2017/05/25 08:40:06 Setting working directory to /tmp/terragrunt-download/nh-GXY9InsiG9x9QeDXnxdMwETM/YNBFcR4X61Vu2vYlFhZFR15I3qI/as-enterprise-vpc
[terragrunt] 2017/05/25 08:40:06 Initializing remote state for the s3 backend
[terragrunt] 2017/05/25 08:40:07 Lock table terragrunt does not exist in DynamoDB. Will need to create it just this first time.
[terragrunt] 2017/05/25 08:40:07 Creating table terragrunt in DynamoDB
[terragrunt] 2017/05/25 08:40:07 Table terragrunt is not yet in active state. Will check again after 10s.
[terragrunt] 2017/05/25 08:40:17 Success! Table terragrunt is now in active state.
[terragrunt] 2017/05/25 08:40:17 Configuring remote state for the s3 backend
[terragrunt] 2017/05/25 08:40:17 Running command: terraform init -backend-config=encrypt=true -backend-config=lock_table=terragrunt -backend-config=bucket=uiuc-tech-services-as-test -backend-config=key=terraform/us-east-2/vpc/terraform.tfstate -backend-config=region=us-east-2
Deprecation warning: This environment is configured to use legacy remote state.
Remote state changed significantly in Terraform 0.9. Please update your remote
state configuration to use the new 'backend' settings. For now, Terraform
will continue to use your existing settings. Legacy remote state support
will be removed in Terraform 0.11.

You can find a guide for upgrading here:

https://www.terraform.io/docs/backends/legacy-0-8.html


Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your environment. If you forget, other
commands will detect it and remind you to do so if necessary.
[terragrunt] 2017/05/25 08:40:19 Running command: terraform plan
Deprecation warning: This environment is configured to use legacy remote state.
Remote state changed significantly in Terraform 0.9. Please update your remote
state configuration to use the new 'backend' settings. For now, Terraform
will continue to use your existing settings. Legacy remote state support
will be removed in Terraform 0.11.
You can find a guide for upgrading here:

https://www.terraform.io/docs/backends/legacy-0-8.html

Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

data.aws_region.current: Refreshing state...
data.aws_caller_identity.current: Refreshing state...
aws_vpc.vpc: Refreshing state... (ID: vpc-3711a35e)
aws_vpc_peering_connection_accepter.pcx: Refreshing state... (ID: pcx-bb21cfd2)
aws_vpc_endpoint.private-s3: Refreshing state... (ID: vpce-17c8287e)
The Terraform execution plan has been generated and is shown below.
Resources are shown in alphabetical order for quick scanning. Green resources
will be created (or destroyed and then created if an existing resource
exists), yellow resources are being changed in-place, and red resources
will be destroyed. Cyan entries are data sources to be read.

Note: You didn't specify an "-out" parameter to save this plan, so when
"apply" is called, Terraform can't guarantee this is what will execute.

+ aws_vpc.vpc
    assign_generated_ipv6_cidr_block: "false"
    cidr_block:                       "10.224.253.0/24"
    default_network_acl_id:           "<computed>"
    default_route_table_id:           "<computed>"
    default_security_group_id:        "<computed>"
    dhcp_options_id:                  "<computed>"
    enable_classiclink:               "<computed>"
    enable_dns_hostnames:             "true"
    enable_dns_support:               "true"
    instance_tenancy:                 "<computed>"
    ipv6_association_id:              "<computed>"
    ipv6_cidr_block:                  "<computed>"
    main_route_table_id:              "<computed>"
    tags.%:                           "1"
    tags.Name:                        "techservicesastest-vpc"

+ aws_vpc_endpoint.private-s3
    cidr_blocks.#:     "<computed>"
    policy:            "<computed>"
    prefix_list_id:    "<computed>"
    route_table_ids.#: "<computed>"
    service_name:      "com.amazonaws.us-east-2.s3"
    vpc_id:            "${aws_vpc.vpc.id}"

+ aws_vpc_peering_connection_accepter.pcx
    accept_status:             "<computed>"
    accepter.#:                "<computed>"
    auto_accept:               "true"
    peer_owner_id:             "<computed>"
    peer_vpc_id:               "<computed>"
    requester.#:               "<computed>"
    vpc_id:                    "<computed>"
    vpc_peering_connection_id: "pcx-bb21cfd2"

- module.vpc.aws_vpc.vpc

- module.vpc.aws_vpc_endpoint.private-s3

- module.vpc.aws_vpc_peering_connection_accepter.pcx


Plan: 3 to add, 0 to change, 3 to destroy.
```
Here is the contents of a script I wrote to make some state changes:
```
#!/bin/sh

terragrunt state mv module.vpc.aws_caller_identity.current data.aws_caller_identity.current
terragrunt state mv module.vpc.aws_region.current data.aws_region.current
terragrunt state mv module.vpc.aws_vpc.vpc aws_vpc.vpc
terragrunt state mv module.vpc.aws_vpc_endpoint.private-s3 aws_vpc_endpoint.private-s3
terragrunt state mv module.vpc.aws_vpc_peering_connection_accepter.pcx aws_vpc_peering_connection_accepter.pcx
```
and here is the output with errors:
```
> $ ./refactor.sh 
[terragrunt] [/home/ddriddle/src/as-aws/as-aws-test/us-east-2/vpc] 2017/05/25 08:40:44 Running command: terraform --version
[terragrunt] 2017/05/25 08:40:44 Reading Terragrunt config file at /home/ddriddle/src/as-aws/as-aws-test/us-east-2/vpc/terraform.tfvars
[terragrunt] 2017/05/25 08:40:44 Terraform files in /tmp/terragrunt-download/nh-GXY9InsiG9x9QeDXnxdMwETM/YNBFcR4X61Vu2vYlFhZFR15I3qI/as-enterprise-vpc are up to date. Will not download again.
[terragrunt] 2017/05/25 08:40:44 Copying files from /home/ddriddle/src/as-aws/as-aws-test/us-east-2/vpc into /tmp/terragrunt-download/nh-GXY9InsiG9x9QeDXnxdMwETM/YNBFcR4X61Vu2vYlFhZFR15I3qI/as-enterprise-vpc
[terragrunt] 2017/05/25 08:40:44 open /tmp/terragrunt-download/nh-GXY9InsiG9x9QeDXnxdMwETM/YNBFcR4X61Vu2vYlFhZFR15I3qI/as-enterprise-vpc/.terraform/modules/7f243cf8f80c562d50d2599756bd5ad5/.git/objects/pack/pack-b9a3ac6087e03268b21dc694d20ccf89859a98d3.idx: permission denied
[terragrunt] 2017/05/25 08:40:44 Unable to determine underlying exit code, so Terragrunt will exit with error code 1
[terragrunt] [/home/ddriddle/src/as-aws/as-aws-test/us-east-2/vpc] 2017/05/25 08:40:44 Running command: terraform --version
[terragrunt] 2017/05/25 08:40:44 Reading Terragrunt config file at /home/ddriddle/src/as-aws/as-aws-test/us-east-2/vpc/terraform.tfvars
[terragrunt] 2017/05/25 08:40:44 Terraform files in /tmp/terragrunt-download/nh-GXY9InsiG9x9QeDXnxdMwETM/YNBFcR4X61Vu2vYlFhZFR15I3qI/as-enterprise-vpc are up to date. Will not download again.
[terragrunt] 2017/05/25 08:40:44 Copying files from /home/ddriddle/src/as-aws/as-aws-test/us-east-2/vpc into /tmp/terragrunt-download/nh-GXY9InsiG9x9QeDXnxdMwETM/YNBFcR4X61Vu2vYlFhZFR15I3qI/as-enterprise-vpc
[terragrunt] 2017/05/25 08:40:44 open /tmp/terragrunt-download/nh-GXY9InsiG9x9QeDXnxdMwETM/YNBFcR4X61Vu2vYlFhZFR15I3qI/as-enterprise-vpc/.terraform/modules/7f243cf8f80c562d50d2599756bd5ad5/.git/objects/pack/pack-b9a3ac6087e03268b21dc694d20ccf89859a98d3.idx: permission denied
[terragrunt] 2017/05/25 08:40:44 Unable to determine underlying exit code, so Terragrunt will exit with error code 1
```
Here is the output of terragrunt plan with errors:
```
$ terragrunt plan
[terragrunt] [/home/ddriddle/src/as-aws/as-aws-test/us-east-2/vpc] 2017/05/25 09:00:05 Running command: terraform --version
[terragrunt] 2017/05/25 09:00:05 Reading Terragrunt config file at /home/ddriddle/src/as-aws/as-aws-test/us-east-2/vpc/terraform.tfvars
[terragrunt] 2017/05/25 09:00:05 Terraform files in /tmp/terragrunt-download/nh-GXY9InsiG9x9QeDXnxdMwETM/YNBFcR4X61Vu2vYlFhZFR15I3qI/as-enterprise-vpc are up to date. Will not download again.
[terragrunt] 2017/05/25 09:00:05 Copying files from /home/ddriddle/src/as-aws/as-aws-test/us-east-2/vpc into /tmp/terragrunt-download/nh-GXY9InsiG9x9QeDXnxdMwETM/YNBFcR4X61Vu2vYlFhZFR15I3qI/as-enterprise-vpc
[terragrunt] 2017/05/25 09:00:05 open /tmp/terragrunt-download/nh-GXY9InsiG9x9QeDXnxdMwETM/YNBFcR4X61Vu2vYlFhZFR15I3qI/as-enterprise-vpc/.terraform/modules/7f243cf8f80c562d50d2599756bd5ad5/.git/objects/pack/pack-b9a3ac6087e03268b21dc694d20ccf89859a98d3.idx: permission denied
[terragrunt] 2017/05/25 09:00:05 Unable to determine underlying exit code, so Terragrunt will exit with error code 1
```
On second look, I think the problem is that I have an old .terraform state file in my directory. After removing `/tmp/terragrunt-download` and `.terragrunt` I now longer receive these errors. If it makes sense it would be great if terragrunt either did not copy over .terraform folders or warned about them.

**Comment by brikis98** [comment]
Ah, you're right. The [CopyFolderContents method](https://github.com/gruntwork-io/terragrunt/blob/master/util/file.go#L119) should skip folders that start with `.`. PRs welcome!

**Comment by brikis98** [comment]
This was fixed a while back by #236.
